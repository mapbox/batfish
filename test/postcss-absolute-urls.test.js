'use strict';

const postcss = require('postcss');
const postcssAbsoluteUrls = require('../src/node/postcss-absolute-urls');

describe('postcssAbsoluteUrls', () => {
  test('works', () => {
    const css = `
      @font-face {
        font-family: "Open Sans";
        src: url("../fonts/OpenSans-Regular-webfont.woff2") format("woff2"),
              url("/fonts/OpenSans-Regular-webfont.woff") format("woff");
      }
      body {
        background: url('data:blergh');
      }
    `;

    return postcss()
      .use(
        postcssAbsoluteUrls({
          stylesheetUrl: 'https://www.fake-place.com/path/to/style.css'
        })
      )
      .process(css, { from: undefined })
      .then((result) => {
        expect(result.css).toMatchSnapshot();
      });
  });
});
