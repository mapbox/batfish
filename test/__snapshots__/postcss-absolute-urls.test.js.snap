// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postcssAbsoluteUrls works 1`] = `
"
      @font-face {
        font-family: \\"Open Sans\\";
        src: url(\\"https://www.fake-place.com/path/fonts/OpenSans-Regular-webfont.woff2\\") format(\\"woff2\\"),
              url(\\"https://www.fake-place.com/fonts/OpenSans-Regular-webfont.woff\\") format(\\"woff\\");
      }
      body {
        background: url('data:blergh');
      }
    "
`;
